client/state-db/src/noncanonical.rs

```diff
@@ -38,6 +38,7 @@ pub struct NonCanonicalOverlay<BlockHash: Hash, Key: Hash> {
 	// would be deleted but kept around because block is pinned, ref counted.
 	pinned: HashMap<BlockHash, u32>,
 	pinned_insertions: HashMap<BlockHash, (Vec<Key>, u32)>,
+	last_canon_pinned: Option<BlockHash>,
```

Comment:
Could `last_canonicalized` been used instead? Might be a bit less explicit though.

Reply:
`last_canonicalized` is set on startup with the block that's already in the DB and can't be pinned. There needs to be a separate value to track what has actually been pinned.

Reply:
I missed that, having a separate field is certainly more explicit. Could almost make it a local variable, but then some unpin will run on these startup value too (which may not be an issue here), but think this is more robust (having the local field).